packages/command-palette/src/utils.ts

```diff
@@ -0,0 +1,135 @@
+import {
+	PLAN_MIGRATION_TRIAL_MONTHLY,
+	PLAN_ECOMMERCE_TRIAL_MONTHLY,
+	PLAN_HOSTING_TRIAL_MONTHLY,
+} from '@automattic/calypso-products';
+import { Falsy } from 'utility-types';
+import { SiteExcerptData, SiteExcerptNetworkData } from './site-excerpt-types';
+
+export function isCustomDomain( siteSlug: string | null | undefined ): boolean {
+	if ( ! siteSlug ) {
+		return false;
+	}
+	return ! siteSlug.endsWith( '.wordpress.com' ) && ! siteSlug.endsWith( '.wpcomstaging.com' );
+}
+
+export const isNotAtomicJetpack = ( site: SiteExcerptNetworkData ) => {
+	return site.jetpack && ! site?.is_wpcom_atomic;
+};
+
+export const isP2Site = ( site: SiteExcerptNetworkData ) => {
+	return site.options?.is_wpforteams_site;
+};
+
+export const isMigrationTrialSite = ( site: SiteExcerptNetworkData ) => {
+	return site?.plan?.product_slug === PLAN_MIGRATION_TRIAL_MONTHLY;
+};
+
+export const isHostingTrialSite = ( site: SiteExcerptNetworkData ) => {
+	return site?.plan?.product_slug === PLAN_HOSTING_TRIAL_MONTHLY;
+};
+
+export const isECommerceTrialSite = ( site: SiteExcerptNetworkData ) => {
+	return site?.plan?.product_slug === PLAN_ECOMMERCE_TRIAL_MONTHLY;
+};
+
+export const isBusinessTrialSite = ( site: SiteExcerptNetworkData ) => {
+	return isMigrationTrialSite( site ) || isHostingTrialSite( site );
+};
+
+export const isTrialSite = ( site: SiteExcerptNetworkData ) => {
+	return isBusinessTrialSite( site ) || isECommerceTrialSite( site );
+};
+
+export const siteDefaultInterface = ( site: SiteExcerptNetworkData ) => {
+	return site?.options?.wpcom_admin_interface;
+};
```

Comment:
Maybe we could implement these functions in a different package consumed by the command-palette and calypso?